modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/utilities/dataRenderers.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import dataRenderers from '../data_renderers/index';
+import {getJsModule} from './modules';
+
+export function getDataRendererById(id) {
+	const dataRenderer = dataRenderers[id];
+
+	if (!dataRenderer) {
+		throw new Error(`No data renderer found with id "${id}"`);
+	}
+
+	return dataRenderer;
+}
+
+export const fetchedContentRenderers = [];
+
+export function getDataRendererByURL(url) {
+	return new Promise((resolve, reject) => {
+		const addedDataRenderer = fetchedContentRenderers.find(
+			(cr) => cr.url === url
```

Comment:
[Abbreviations are bad](https://github.com/liferay/liferay-frontend-guidelines/blob/master/general/naming.md#avoid-abbreviations). Especially so for one like this which already has a widely-used meaning in computing (["carriage return"](https://en.wikipedia.org/wiki/Carriage_return)).